### PR TITLE
openjdk

### PR DESCRIPTION
**Pull request by zschulte**
Adding openjdk.

Modified #9861 and added 11.0.2 and 1.8.

### CODE REVIEW

**Comment by scheibelp** [comment]
I'm inclined to merge https://github.com/spack/spack/pull/9861 (it needs to make one minor edit though). Would you be able to submit this as an addition to that?

**Comment by zschulte** [comment]
That would be fine.  I noticed the other one hadn't been touched in a while which is why I made this one.

**Comment by scheibelp** [comment]
> That would be fine. I noticed the other one hadn't been touched in a while which is why I made this one.

Yes I lost track of that one. It is merged now.

File: var/spack/repos/builtin/packages/openjdk/package.py [hunk 1]
```diff
@@ -28,69 +44,80 @@ class Openjdk(Package):
...
-
+        """
+        Set JAVA_HOME and CLASSPATH.
         CLASSPATH contains the installation prefix for the extension and any
-        other Java extensions it depends on."""
+        other Java extensions it depends on.
+        """
 
-        spack_env.set('JAVA_HOME', self.home)
+        spack_env.set("JAVA_HOME", self.home)
```
[the remark this answers is not in the document]
Comment (zschulte) [reply var/spack/repos/builtin/packages/openjdk/package.py]: I've removed the formatting changes.  I usually run my stuff through black (github.com/ambv/black) to keep formatting consistent but understand the desire to keep formatting changes separate from function changes.

**Comment by zschulte** [comment]
It looks like the url for 1.8 broke.  I'm looking for an alternative.

**Comment by zschulte** [comment]
> It looks like the url for 1.8 broke. I'm looking for an alternative.

I've pointed the 1.8 version at github.com/AdoptOpenJDK/openjdk8-binaries.  It seems like a much more stable release environment for java 1.8.  At some point it could probably be put into a list_url.  It may also make sense to make the AdoptOpenJDK project the source for the other versions as well.

**Comment by zschulte** [comment]
It looks like the link I had used before wasn't actually to the binary.  I have fixed the link to point at the x64 linux binary.  There are many more in that same repo for things like Macs and such.  It would probably be good to have either it actually build from source or logic to determine which binary is needed.

**Pull request merged**